tests/unit_tests/utils/test_general.py

```diff
@@ -133,3 +134,26 @@ def test_program_is_executable():
     # (assume 'ls' exist on any system the test is running)
     assert gen.program_is_executable("ls") is not None
     assert gen.program_is_executable("this_program_probably_does_not_exist") is None
+
+
+def test_change_dict_keys_case():
+
+    _upper_dict = {
+        "REFERENCE": {"VERSION": "0.1.0"},
+        "ACTIVITY": {"NAME": "submit", "ID": "84890304", "DESCRIPTION": "Set data"},
+    }
+    _lower_dict = {
+        "reference": {"version": "0.1.0"},
+        "activity": {"name": "submit", "id": "84890304", "description": "Set data"},
+    }
+    _no_change_dict_upper = gen.change_dict_keys_case(copy(_upper_dict), False)
+    assert _no_change_dict_upper == _upper_dict
+
+    _no_change_dict_lower = gen.change_dict_keys_case(copy(_upper_dict), True)
```

Comment:
I think that in this case the input should be `_lower_dict` and `True`

Reply:
Yes, good catch!